model/context.go

```diff
@@ -12,32 +12,34 @@ import (
 // function, all the other applicable fields in the Context can
 // inferred and populated from the id of any one of the fields.
 type Context struct {
-	Task       *task.Task
-	Build      *build.Build
-	Version    *Version
-	Patch      *patch.Patch
-	ProjectRef *ProjectRef
-
-	project *Project
+	Task    *task.Task
+	Build   *build.Build
+	Version *Version
+	Patch   *patch.Patch
+
+	projectRef     *ProjectRef
+	project        *Project
+	invalidProject bool
```

Comment:
It's not clear to me what the purpose of this is. Can't we do JIT validation of the project instead of storing this?